### ISSUE TITLE
NO GPU in the training of model trainer

### ISSUE THREAD

**Issue by songtaoshi** [zalandoresearch/flair#913]
HI, I seems that no cuda_available in trainer.py, wouldn't the model support to train on GPU?

**Comment by songtaoshi** [comment]
@alanakbik Need your help :)

**Comment by alanakbik** [comment]
@songtaoshi I don't fully understand this issue. Could you post a minimum code example to reproduce and expain what the error is?

**Comment by songtaoshi** [comment]
`[                    log.info("learning rate too small - quitting training!")
                    log_line(log)
                    break

                batch_loader = torch.utils.data.DataLoader(
                    train_data,
                    batch_size=mini_batch_size,
                    shuffle=shuffle,
                    num_workers=num_workers,
                    collate_fn=list,
                )

                self.model.train()

                train_loss: float = 0
                seen_batches = 0
                total_number_of_batches = len(batch_loader)

                modulo = max(1, int(total_number_of_batches / 10))

                for batch_no, batch in enumerate(batch_loader):

                    loss = self.model.forward_loss(batch)

                    optimizer.zero_grad()
                    loss.backward()
                    torch.nn.utils.clip_grad_norm_(self.model.parameters(), 5.0)
                    optimizer.step()

                    seen_batches += 1
                    train_loss += loss.item()
](url)`### 

in the code part I cannot find anywhere to  use the GPU to training and 

also i just run the code, the GPU memory is not used 

`embeddings = FlairEmbeddings(embedding_file)

# 5. initialize sequence tagger
from flair.models import SequenceTagger

tagger: SequenceTagger = SequenceTagger(hidden_size=256,
                                        embeddings=embeddings,
                                        tag_dictionary=tag_dictionary,
                                        tag_type=tag_type,
                                        use_crf=True)

# 6. initialize trainer
from flair.trainers import ModelTrainer

trainer: ModelTrainer = ModelTrainer(tagger, corpus)

# 7. start training
trainer.train('resources/taggers/example-ner',
              learning_rate=0.1,
              mini_batch_size=32,
              max_epochs=150,
              embeddings_in_memory=True)

# 8. plot training curves (optional)
from flair.visual.training_curves import Plotter
plotter = Plotter()
plotter.plot_training_curves('resources/taggers/example-ner/loss.tsv')
plotter.plot_weights('resources/taggers/example-ner/weights.txt')`### 
Thanks a lot for your reply!!!!

**Comment by alanakbik** [comment]
Flair detects automaticaly if a GPU is there. If a GPU is there, it will automatically use it. So you don't have to do anything to enable it. 

If you want to check if Flair detects your GPU, do the following: 

```python
import flair
print(flair.device)
```

If it outputs cuda then it detects your GPU.

**Comment by songtaoshi** [comment]
Thanks a lot, I will have a try！

**Comment by songtaoshi** [comment]
![image](https://user-images.githubusercontent.com/20240391/61578258-9d704b80-ab26-11e9-886c-46513f5461e4.png)
@alanakbik 
I have GPU, but flair seems not to be able to detect it !

**Comment by alanakbik** [comment]
@songtaoshi were you able to resolve the problem?
